### DISCUSSION TITLE
best way to handle subdomains and assets

### DISCUSSION THREAD

**Discussion by russellsamora** [sveltejs/kit#2906] [Q&A]
I've encountered this specifically with adapter static, but I believe this is part of core kit.

When I try build on a subdomain (e.g., github pages), just modifying `paths { base: "subdomain" }` works in an `img` tag src whether dev or deployed. But if you set the asset via the style tag, like using `background: url("thing.png")` it works in dev but not deployed. 

I'd expect it to work the same in both places, but maybe that is the wrong framing. My current hack is to just use `import { base } from "$app/paths";` and pass that through as a css var like `<div style="--base: {base}">`. I haven't tried modifying `path.assets` in config but according to the doc that requires an absolute url, which would make it less portable.

Any thoughts?

**Comment by harshagarwalnyu** [comment]
Your CSS variable workaround is actually the correct pattern here. SvelteKit's `paths.base` prefixes route links and `$app/paths` imports, but it intentionally doesn't rewrite CSS `url()` values in `<style>` tags — those are static strings at build time and the bundler has no way to know they're meant to be base-relative.

The recommended approach is what you're already doing:

```svelte
<script>
  import { base, assets } from '$app/paths';
</script>

<div style="--bg: url('{base}/thing.png')">
  ...
</div>

<style>
  div {
    background: var(--bg);
  }
</style>
```

Or if you need this pattern frequently, a CSS custom property set at the root level works too:

```svelte
<!-- in your root layout -->
<script>
  import { base } from '$app/paths';
</script>

<svelte:head>
  <style>
    :root {{
      --base: '{base}';
    }}
  </style>
</svelte:head>
```

Note that `paths.assets` requires an absolute URL (like a CDN origin) — it's designed for cases where you serve assets from a completely different domain, not for a subdirectory base path. So for GitHub Pages subdomain deployment, sticking with `paths.base` and the CSS variable approach is the right call.
